core/store/src/trie/trie_storage.rs

```diff
@@ -233,7 +233,12 @@ pub struct TrieCache(pub(crate) Arc<Mutex<TrieCacheInner>>);
 
 impl TrieCache {
     pub fn new(config: &TrieConfig, shard_uid: ShardUId, is_view: bool) -> Self {
-        let total_size_limit = config.shard_cache_total_size_limit(shard_uid, is_view);
+        let total_size_limit = config
+            .shard_cache_config
```

Comment:
We should take `view_shard_cache_config` here and below if `is_view` is true, and possibly have a test checking that we take correct values for view and regular caches.

Reply:
Good catch! Fixed and added a test.